### PR TITLE
Survival crafting stuff

### PR DESCRIPTION
**Pull request by Very-Soft**
Stuff! For surviving in the wilderness.

### **I've made it so:** 
you can take rocks (flint) out of dirt
you can get sticks from trees
you can make a wooden shovel
you can make flint knives
you can make flint axes
you can make flint spears
you can make empty dirt turf into a soil grow plot
you can use a shovel to destroy an empty soil grow plot
you can get certain seeds from cutting grass
you can make clothing from fur/wool/leather

I also made the Teppi, which, are a source of relatively unlimited meat, bone, fur, leather, and companionship.

FOR NOW THAT IS ALL I AM DOING!!! I will revisit the rest of this later.

### **I want to make:**
bowl (for containing reagents)
pestle (for grinding things in a bowl)
a way to get seeds without machines (so you can farm)
a way to get nutrition to plants without machines 
using bonfires to cook some simple foods (roast meat, jerky hard tack) (because raw food is lame)
a way to get water (for drinking, and for plants)

### **If you have other ideas for this, please feel free to suggest them in the comments, but keep it to the theme here of natural stuff you can get without machines or electronics. Nature survival type stuff.**

Most of this won't really be too useful to your average spaceman. But some of us quite like to go get lost out in the wilds and such, and we don't have the luxuries of vending machines and robots that can grow our food for us.

BASICALLY, I have had several ideas that would BENEFIT from there being more stuff you could do with and in a natural environment. I've had events in the past that would have been way more fun and engaging if people were able to do more things in the more natural environments. 

I won't get there with this project, but eventually I personally would really like it if the crew could go from naked, having nothing at all, to being able to MAKE all the things that you can play with an use if you have enough time and energy. Not something you'd necessarily -want- to do just on your own, but if all the pieces are there to make that possible, then it could let us have a lot more freedom with setting up events and away maps, and empower players to make stuff where there isn't stuff!

### CODE REVIEW

**Comment by FoxDoodKMG** [comment]
First, some ideas for your stuff.
A way to get seeds without machines, scavenge from dirt, knife that cuts food into seeds.
A way to get nutrition to plants, a fertilizer bucket, fill it with organic material and water.
A way to get bonfires to cook simple foods, cooking rock, will cook food if next to an active bonfire.
A way to get water, wood barrel that works as a water tank, slowly filling up with tiny amounts of water over time while outside.
Now some wants of my own.
Craft-able bows and arrows, needs wood and refined string, made from a sizable amount of cloth
Primitive surgical tools, made from stone and bone.
Clothing made of leaves and vines.
Sorry if I misunderstood.

**Comment by KillianKirilenko** [comment]
> cloth and leather clothing (in case you get mood debuffs from wearing tattered clothing or are otherwise naked and unable to get replacements)

mood debuffs? what is this, rimworld? 😛

#9287 should have you covered though; butchering was made more modular a while ago and now you can get hides & turn it into leather. you just need something sharp, a water source, and a fire, by the sounds of things.

I'd add some way to get bandages to reduce/stop bleeding. use a knife or something else sharp on cloth or other fabrics.

Spearcrafting can be adapted from existing spearcrafting (rod + wires + blade => haft + bindings + blade) pretty easily I'd think.

**Comment by TheDavestDave** [comment]
I feel like knifes and spears already being easy to make might muddle things up but please please please. Can I cook meats on bonfires

**Comment by TheDavestDave** [comment]
oh and you can make wooden bowls (I think) if you have a tree and a knife just takes some work. I think the knife can chop the tree down into logs, then chop the logs into planks, that you can craft into a bowl

**Comment by Very-Soft** [comment]
> I feel like knifes and spears already being easy to make might muddle things up but please please please. Can I cook meats on bonfires

Well the thing with that is, you can make knives and spears if you have the materials for them, but, those are typically made of materials you need other tools for. I want you to be able to make crude versions of them out of stuff you can find.

**Comment by SigholtStarsong** [comment]
I'd suggest pemmican if we're doing survival foods. It's a quintessential provision for that sort of thing.

**Comment by Ketrai** [comment]
Knapping flint together to make toolheads that can be combined with sticks. dig a pit to fill with wood. Burn it, cover it, and you get charcoal. The pit can then be re-used to fire clay pottery. (you can literally just make this an 'oven' via aurora cooking, then give it the appropriate recipes). Compost bin that you can toss other food into or meats to get fertiliser. Ideally the ratio is like, grow plant, harvest once, use 2/3ds for fertiliser and seeds. Eat the leftover veggie. It should be far from the same efficiency wise as using a tray and a proper seed extractor. 

A small chance to find animals cutting grass could be nice. Especially if you can make proper fences out of wood & sticks.

**Comment by Ketrai** [comment]
Also there should be some kind of knowledge system. It makes sense for the crew to be able to claw their way back up to technology. But maybe not everyone for say like tribal event characters.

**Review by Novacat — APPROVED** [review]
Was tested, seems to be fine, I give my approval!

File: code/game/turfs/simulated/outdoors/survival_action_vr.dm [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+		if(locate(/obj) in src)
+			to_chat(user, "<span class='notice'>The [name] isn't clear.</span>")
+			return
+		else
+			var/choice= tgui_alert(user, "Do you want to build a growplot out of the dirt?", "Build growplot?" , list("Yes", "No"))
+			if(!choice||choice=="No")
+				return
+			user.visible_message("[user] starts piling up [src]...", runemessage = "piles up [src]")
+			if(do_after(user, 5 SECONDS, exclusive = TASK_USER_EXCLUSIVE))
+				new /obj/machinery/portable_atmospherics/hydroponics/soil(src)
```
Comment (Arokha) [inline code/game/turfs/simulated/outdoors/survival_action_vr.dm:24]: Optional: `if(!Adjacent(user)) return ..()` instead so it exits early rather than having it all indented

File: code/game/turfs/simulated/outdoors/survival_action_vr.dm [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+			if(prob(5))
+				var/obj/item/stack/material/flint/R = new(get_turf(src))
+				to_chat(user, "<span class='notice'>You found some [R]</span>")
+				R.amount = rand(1,4)
+				R.pixel_x = rand(-6,6)
+				R.pixel_y = rand(-6,6)
+			else
+				to_chat(user, "<span class='notice'>You didn't find anything...</span>")
+	else
+		..()
```
Comment (Arokha) [inline code/game/turfs/simulated/outdoors/survival_action_vr.dm:42]: Suggested: Probably should be `return ..()`
Repeat for all the attackbys on your turfs.

Honestly this copypaste could be moved to a proc on /turf/simulated/floor where you do all the stuff.
Just a whole
```
if(istype(O, /obj/item/weapon/shovel))
   do_survival_thing(user, O) // or whatever
```

File: code/modules/hydroponics/trays/tray_soil.dm [hunk 3]
```diff
@@ -10,6 +10,16 @@
...
+			var/choice= tgui_alert(user, "Do you want to destroy the growplot?", "Destroy growplot?" , list("Yes", "No"))
+			if(!choice||choice=="No")
+				return
+			user.visible_message("[user] starts dispersing the [src]...", runemessage = "disperses the [src]")
+			if(do_after(user, 5 SECONDS, exclusive = TASK_USER_EXCLUSIVE))
+				qdel(src)
+		else
+			to_chat(user, "<span class='notice'>There is something growing here.</span>")
 	else
 		..()
```
Comment (Arokha) [inline code/modules/hydroponics/trays/tray_soil.dm:24]: Suggested: Probably `return ..()`

File: code/modules/mob/living/simple_mob/subtypes/animal/alien animals/teppi.dm [hunk 4]
```diff
@@ -0,0 +1,1195 @@
...
+	var/inherit_colors = FALSE
+	var/teppi_wool = FALSE
+	var/amount_grown = 0
+	var/teppi_adult = TRUE
+	var/friend_zone //where friends go when we eat them
+//	var/teppi_id //This is all for anti-incest business, which I might finish eventually, but am not sure if it's really deisrable right now.
+//	var/mom_id
+//	var/dad_id
+	var/baby_countdown = 0
+	var/breedable = FALSE
```
Comment (Arokha) [inline code/modules/mob/living/simple_mob/subtypes/animal/alien animals/teppi.dm:115]: owo

File: code/modules/mob/living/simple_mob/subtypes/animal/alien animals/teppi.dm [hunk 5]
```diff
@@ -0,0 +1,1195 @@
...
+		return ..()
+	if(istype(O, /obj/item/weapon/holder))
+		return ..()
+	if(user.a_intent != I_HELP) //be gentle
+		if(resting)
+			lay_down()
+		handle_affinity(user, -5)
+		user.visible_message(user, "<span class='notice'>\The [user] hits \the [src] with \the [O]. \The [src] grumbles at \the [user].</span>","<span class='notice'>You hits \the [src] with \the [O]. \The [src] grumbles at you.</span>")
+		playsound(src, 'sound/weapons/tap.ogg', 50, 1, -1)
+		return..()
```
Comment (Arokha) [inline code/modules/mob/living/simple_mob/subtypes/animal/alien animals/teppi.dm:497]: Pls change: Needs space, not even sure why this compiles. Byond may be more forgiving than I am

File: code/modules/mob/living/simple_mob/subtypes/animal/alien animals/teppi.dm [hunk 6]
```diff
@@ -0,0 +1,1195 @@
...
+			user.visible_message("<span class='notice'>\The [user] shears \the [src] with \the [O].</span>","<span class='notice'>You shear \the [src] with \the [O].</span>")
+			if(do_after(user, sheartime, exclusive = TASK_USER_EXCLUSIVE, target = src))
+				amount_grown = rand(0,250)
+				var/obj/item/stack/material/fur/F = new(get_turf(user))
+				F.amount = rand(10,15)
+				F.color = marking_color
+				teppi_wool = FALSE
+				update_icon()
+				handle_affinity(user, 5)
+				return
```
Comment (Arokha) [inline code/modules/mob/living/simple_mob/subtypes/animal/alien animals/teppi.dm:578]: Suggestion: Duplicated code could be moved to a `/proc/shear(user, tool)` proc or somesuch.

File: code/modules/mob/living/simple_mob/subtypes/animal/alien animals/teppi.dm [hunk 7]
```diff
@@ -0,0 +1,1195 @@
...
+		if(bruteloss > 0)
+			adjustBruteLoss(-2)
+		else if(fireloss > 0)
+			adjustFireLoss(-0.5)
+		nutrition -= 5
+		heal_countdown = 5
+
+/mob/living/simple_mob/vore/alienanimals/teppi/proc/do_breeding(not_hungy)
+	if(!breedable || !not_hungy || prevent_breeding)
+		return
```
Comment (Arokha) [inline code/modules/mob/living/simple_mob/subtypes/animal/alien animals/teppi.dm:731]: Suggestion: Seems pointless to pass `not_hungy` in, because in both procs you just use it to decide not to do anything on the first line.
Just do
```
if(not_hungy)
    do_breeding()
    do_healing()
```
and remove the not_hungy checks inside the procs (and remove it from the arglist on the procs)

File: code/modules/mob/living/simple_mob/subtypes/animal/alien animals/teppi.dm [hunk 8]
```diff
@@ -0,0 +1,1195 @@
...
+	if(client)	//Player controlled teppi get a verb, so just do the countdown
+		if(baby_countdown > 0)
+			baby_countdown -= 1
+		return
+	if(baby_countdown > 0)
+		baby_countdown -= 1
```
Comment (Arokha) [inline code/modules/mob/living/simple_mob/subtypes/animal/alien animals/teppi.dm:737]: Comment: You can just do `baby_countdown--` which subtracts one, and is *stylish*.

You can also do this on one line with `baby_countdown = max(baby_countdown-1, 0)` which sets it to the higher of baby_countdown-1 or 0 (which would be higher if it became a negative number). Your way might be faster though, I'm not really sure.

File: code/modules/mob/living/simple_mob/subtypes/animal/alien animals/teppi.dm [hunk 9]
```diff
@@ -0,0 +1,1195 @@
...
+/mob/living/simple_mob/vore/alienanimals/teppi/proc/teppi_sound()
+	if(!teppi_adult || client)
+		return
+	if(resting)
+		return
+	playsound(src, pick(teppi_sound), 75, 1)
+
+//Handles both growing up from a baby and also passing parent details to new babies.
+/mob/living/simple_mob/vore/alienanimals/teppi/New(newloc, teppi1, teppi2)
+	GLOB.teppi_count += 1
```
Comment (Arokha) [inline code/modules/mob/living/simple_mob/subtypes/animal/alien animals/teppi.dm:764]: Comment: Could also do `GLOB.teppi_count++` which is *stylish*.

File: code/modules/mob/living/simple_mob/subtypes/animal/alien animals/teppi.dm [hunk 10]
```diff
@@ -0,0 +1,1195 @@
...
+	if(teppi1 && !teppi2)
+		inherit_from_baby(teppi1)
+	else if (teppi1 && teppi2)
+		inherit_from_parents(teppi1, teppi2)
+	..()
+
+/mob/living/simple_mob/vore/alienanimals/teppi/Destroy()
+	GLOB.teppi_count -= 1
```
Comment (Arokha) [inline code/modules/mob/living/simple_mob/subtypes/animal/alien animals/teppi.dm:772]: Comment: Could also do `GLOB.teppi_count--` which is *stylish*.

File: code/modules/mob/living/simple_mob/subtypes/animal/alien animals/teppi.dm [hunk 11]
```diff
@@ -0,0 +1,1195 @@
...
+	friend_zone = null
+	active_ghost_pods -= src
+	ai_holder.leader = null
+	return ..()
+
+/mob/living/simple_mob/vore/alienanimals/teppi/Login()
+	..()
+	to_chat(src, "<span class='notice'>While you may have access to galactic common, this is purely meant for making it so you can understand people in an OOC manner, for facilitating roleplay. You almost certainly should not be speaking to people or roleplaying as though you understand everything everyone says perfectly, but it's not unreasonable to be able to intuit intent and such through people's tones when they speak. Teppi are kind of smart, but they are animals, and should be roleplayed as such.</span> <span class='warning'>ADDITIONALLY, you have the ability to produce offspring if you're well fed enough every once in a while, and the ability to disable this from happening to you. These verbs exist for to preserve the mechanical functionality of the mob you are playing. You should be aware of your surroundings when you use this verb, and NEVER use it to prefbreak or be disruptive. If in doubt, don't use it.</span> <span class='notice'>Also, to note, AI Teppi will never initiate breeding with player Teppi.</span>")
```
Comment (Arokha) [inline code/modules/mob/living/simple_mob/subtypes/animal/alien animals/teppi.dm:780]: Pls change: No need to do this, there is already a var on simple_mobs called `player_msg` that is printed to players when they log in. It's null by default and prints nothing. Just put your message in there on teppi.

File: code/modules/mob/living/simple_mob/subtypes/animal/alien animals/teppi.dm [hunk 12]
```diff
@@ -0,0 +1,1195 @@
...
+		vore_selected = friend_zone
+		..()
+		vore_selected = tumby
+		return
+	else if(current_affinity <= -50)
+		vore_selected.digest_mode = DM_DIGEST
+	else 
+		vore_selected.digest_mode = DM_DRAIN
+	..()
+	ai_holder.busy = FALSE
```
Comment (Arokha) [inline code/modules/mob/living/simple_mob/subtypes/animal/alien animals/teppi.dm:814]: Comment: This seems particularly annoying to have to do this way. Maybe in the future there can be a better way to just pick a belly to use.

File: code/modules/mob/living/simple_mob/subtypes/animal/alien animals/teppi.dm [hunk 13]
```diff
@@ -0,0 +1,1195 @@
...
+	prey.stop_pulling()
+	if(current_affinity >= 50)
+		belly = friend_zone
+		return ..()
+	if(current_affinity <= -50)
+		vore_selected.digest_mode = DM_DIGEST
+	else 
+		vore_selected.digest_mode = DM_DRAIN
+	..()
+	ai_holder.busy = FALSE
```
Comment (Arokha) [inline code/modules/mob/living/simple_mob/subtypes/animal/alien animals/teppi.dm:831]: Comment: As above

File: code/modules/mob/living/simple_mob/subtypes/animal/alien animals/teppi.dm [hunk 14]
```diff
@@ -0,0 +1,1195 @@
...
+	set name = "Produce Offspring"
+	set category = "Abilities"
+	set desc = "You can have babies if the conditions are right."
+	if(prevent_breeding)
+		to_chat(src, "<span class='notice'>You have elected to not participate in breeding mechanics, and so cannot complete that action.</span>")
+		return
+	if(!teppi_warned)
+		to_chat(src, "<span class='danger'>Be aware of your surroundings when using this verb. If you use this to be disruptive or prefbreak people, you are likely to eat a ban. If whoever's tending the teppi is trying to make more babies, or you're alone, or playing with other people who you know are into it, then sure. You should not however, for example, drag another teppi to the bar (or any public place) and drop a baby in the middle of the floor. If you're not sure if it's okay to do where you are, with whoever's around, it probably isn't. This is intended to preserve the mechanical utility of the mob you are playing as, not as a scene tool.</span>")
+		teppi_warned = TRUE
+		return
```
Comment (Arokha) [inline code/modules/mob/living/simple_mob/subtypes/animal/alien animals/teppi.dm:989]: Comment: Doesn't account for multiple people playing the same teppi over time, but that's probably pedantic. I handled that on the translocator beacons but those are much more likely to be touched by more than one person, so eh!

File: code/modules/mob/living/simple_mob/subtypes/animal/alien animals/teppi.dm [hunk 15]
```diff
@@ -0,0 +1,1195 @@
...
+	set desc = "You can toggle whether or not you can produce offspring."
+	if(!prevent_breeding)
+		to_chat(src, "<span class='notice'>You disable breeding.</span>")
+		prevent_breeding = TRUE
+	else
+		to_chat(src, "<span class='notice'>You enable breeding.</span>")
+		prevent_breeding = FALSE
+		
+///////////////////AI Things////////////////////////
+//Thank you very much Aronai <3
```
Comment (Arokha) [inline code/modules/mob/living/simple_mob/subtypes/animal/alien animals/teppi.dm:1033]: uwu

File: code/modules/mob/living/simple_mob/subtypes/animal/alien animals/teppi.dm [hunk 16]
```diff
@@ -0,0 +1,1195 @@
...
+			if(their_affinity >= 25 || their_affinity <= -10)
+				.[M] = affinity[M.real_name]
+	// Sort the list (timsort default sort comperator is numeric ascending, so highest affinity will be last in the list)
+	sortTim(., associative = TRUE)
+
+/datum/ai_holder/simple_mob/teppi/handle_wander_movement()
+	var/mob/living/simple_mob/vore/alienanimals/teppi/tepholder = holder
+	if(tepholder.resting)
+		if(prob(5))
+			tepholder.lay_down()
```
Comment (Arokha) [inline code/modules/mob/living/simple_mob/subtypes/animal/alien animals/teppi.dm:1054]: Suggestion: Could be a single if()

File: code/modules/mob/living/simple_mob/subtypes/animal/alien animals/teppi.dm [hunk 17]
```diff
@@ -0,0 +1,1195 @@
...
+/datum/ai_holder/simple_mob/teppi/on_hear_say(mob/living/speaker, message)
+	var/mob/living/simple_mob/vore/alienanimals/teppi/T = holder
+	if(holder.client)
+		return
+	if(!speaker.client)
+		return
+	if(!T.teppi_adult)
+		return
+	var/speaker_affinity = T.affinity[speaker.real_name]
+	if(findtext(message, "lets go") || findtext(message, "come teppi") || findtext(message, "come [holder.name]"))
```
Comment (Arokha) [inline code/modules/mob/living/simple_mob/subtypes/animal/alien animals/teppi.dm:1112]: Comment: You could probably fix the apostrophe problem by doing message = `html_decode(message)` (or maybe it's urlencoded, I dunno so in that case url_decode(message))

File: code/modules/mob/living/simple_mob/subtypes/animal/alien animals/teppi.dm [hunk 18]
```diff
@@ -0,0 +1,1195 @@
...
+
+/mob/living/simple_mob/vore/alienanimals/teppi/mutant
+
+/mob/living/simple_mob/vore/alienanimals/teppi/mutant/New()
+	teppi_mutate = TRUE
+	. = ..()
+
+//Custom teppi colors! For funzies.
+
+/mob/living/simple_mob/vore/alienanimals/teppi/cass
```
Comment (Arokha) [inline code/modules/mob/living/simple_mob/subtypes/animal/alien animals/teppi.dm:1149]: Comment: You don't actually need to do the blank type definitions. The New() on the type implies them into existence.

This can be especially annoying sometimes like on z-shadows when I forgot to remove some proc on them, so they continued existing just because there was a single proc defined on their type despite not being mentioned anywhere else in code.

**Review by Arokha — CHANGES_REQUESTED** [review]
Couple smol things

File: code/modules/mob/living/simple_mob/subtypes/animal/alien animals/teppi.dm (hunk 11 above)
Comment (Very-Soft) [reply code/modules/mob/living/simple_mob/subtypes/animal/alien animals/teppi.dm:780]: For this one, I did this in two steps because player_msg doesn't seem to allow any formatting. I put ICish info in the player_msg, and OOCish info here, because it can be formatted.

Comment (Arokha) [reply code/modules/mob/living/simple_mob/subtypes/animal/alien animals/teppi.dm:780]: It should allow formatting, it's just in a to_chat like this is. Test again? Make sure to include the spans and stuff from your current one.

File: code/modules/mob/living/simple_mob/subtypes/animal/alien animals/teppi.dm (hunk 16 above)
Comment (Very-Soft) [reply code/modules/mob/living/simple_mob/subtypes/animal/alien animals/teppi.dm:1054]: If I do this as a single if then the teppi wanders while it's resting. This tells it to return instead of wandering when it's resting, but also check to see if it's time to wake up.

File: code/modules/mob/living/simple_mob/subtypes/animal/alien animals/teppi.dm (hunk 17 above)
Comment (Very-Soft) [reply code/modules/mob/living/simple_mob/subtypes/animal/alien animals/teppi.dm:1112]: This worked with html_decode I think!

File: code/modules/mob/living/simple_mob/subtypes/animal/alien animals/teppi.dm (hunk 14 above)
Comment (Very-Soft) [reply code/modules/mob/living/simple_mob/subtypes/animal/alien animals/teppi.dm:988]: This is true, though, I'm not super sure how to resolve that... hmm. Maybe Login should set teppi_warned to false?

File: code/modules/mob/living/simple_mob/subtypes/animal/alien animals/teppi.dm (hunk 11 above)
Comment (Very-Soft) [reply code/modules/mob/living/simple_mob/subtypes/animal/alien animals/teppi.dm]: I fiddled with it and got it working!

**Comment by Very-Soft** [comment]
Did most of your suggestions, and replied to the ones I didn't.

**Comment by KasparoVy** [comment]
Great job!

File: code/modules/mob/living/simple_mob/subtypes/animal/alien animals/teppi.dm (hunk 14 above)
Comment (Arokha) [reply code/modules/mob/living/simple_mob/subtypes/animal/alien animals/teppi.dm:988]: I wouldn't worry too much about it, honestly.

File: code/modules/mob/living/simple_mob/subtypes/animal/alien animals/teppi.dm (hunk 16 above)
Comment (Arokha) [reply code/modules/mob/living/simple_mob/subtypes/animal/alien animals/teppi.dm:1053]: oic, I misread

File: code/modules/mob/living/simple_mob/subtypes/animal/alien animals/teppi.dm (hunk 17 above)
Comment (Arokha) [reply code/modules/mob/living/simple_mob/subtypes/animal/alien animals/teppi.dm]: \o/

**Review by Arokha — APPROVED** [review]
